src/main/kotlin/org/jetbrains/research/libsl/visitors/ExpressionVisitor.kt

```diff
@@ -211,83 +224,86 @@ class ExpressionVisitor(
-                    ctx.text.dropLast(2).toUByte(),
-                    "ub",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "s" -> IntegerLiteral(
-                    ctx.text.dropLast(1).toShort(),
-                    "s",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "us" -> UnsignedInt16Literal(
-                    ctx.text.dropLast(2).toUShort(),
-                    "us",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                null -> IntegerLiteral(
-                    ctx.text.toInt(),
-                    "",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "u" -> UnsignedInt32Literal(
-                    ctx.text.dropLast(1).toUInt(),
-                    "u",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "L" -> IntegerLiteral(
-                    ctx.text.dropLast(1).toLong(),
-                    "L",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "uL" -> UnsignedInt64Literal(
-                    ctx.text.dropLast(2).toULong(),
-                    "uL",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                else -> throw IllegalArgumentException("Incorrect integer suffix")
-            }
+                ctx.text.dropLast(1).toByte(),
+                "b",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+        } else if (num.endsWith("us")) {
+            return UnsignedInt16Literal(
+                ctx.text.dropLast(2).toUShort(),
+                "us",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+        } else if (num.endsWith("s")) {
+            return IntegerLiteral(
+                ctx.text.dropLast(1).toShort(),
+                "s",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+        } else if (num.endsWith("u")) {
+            return UnsignedInt32Literal(
+                ctx.text.dropLast(1).toUInt(),
+                "u",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+        } else if (num.endsWith("ul")) {
+            return UnsignedInt64Literal(
+                ctx.text.dropLast(2).toULong(),
+                "uL",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+        } else if (num.endsWith("l")) {
+            num = num.dropLast(1)
+            return IntegerLiteral(
+                convertBinHexOctToPrimitives(num, "l"),
+                "L",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
         }
+        return IntegerLiteral(
+            convertBinHexOctToPrimitives(num, "i"),
```

Comment:
As I understand it, this implies that there  is only `int32` and `int64` primitives can be expressed via 2-, 8- and 16-bit literals (binary, octal, hexadecimal).
Is this the intended behavior?

Reply:
Yes, I suppose it will be enough... Or not ?